src/java.desktop/share/classes/javax/swing/JViewport.java

```diff
@@ -603,11 +604,15 @@ public final Insets getInsets(Insets insets) {
 
 
     private Graphics getBackingStoreGraphics(Graphics g) {
-        Graphics bsg = backingStoreImage.getGraphics();
-        bsg.setColor(g.getColor());
-        bsg.setFont(g.getFont());
-        bsg.setClip(g.getClipBounds());
-        return bsg;
+        if (g instanceof SunGraphics2D) {
```

Comment:
One thing is the change is not obvious as to why it is being done.
I think we can make the fix more telling by utilizing isPrinting check to make it prominent that the code path is for printing
```suggestion
       if (!SwingUtilities2.isPrinting(g)) {
```
Can you see if it works? And yes, you have to make the method public..

Reply:
Yes, it works, can't use `SwingUtilities2.isPrinting(g)` but can check for `PrintGraphics` instance which is done in `SwingUtilities2.isPrinting(g)` .. Thank you for the input...

Reply:
Why cant you use isPrinting?

Reply:
It's not public..

Reply:
> And yes, you have to make the method public..

Thats what I told already..

Reply:
Ok, guess I missed it..

Reply:
Updated.